tests/unit/package_managers/test_gomod.py

```diff
@@ -244,6 +248,136 @@ def test_resolve_gomod(
     )
 
 
+@pytest.mark.parametrize(
+    "cgo_disable, force_gomod_tidy",
+    (
+        pytest.param(False, False, id="cgo_disabled_and_dont_force_tidy"),
+        pytest.param(True, False, id="cgo_enabled_and_dont_force_tidy"),
+        pytest.param(False, True, id="cgo_disabled_and_force_tidy"),
+        pytest.param(True, True, id="cgo_enabled_and_force_tidy"),
+    ),
+)
+@mock.patch("cachi2.core.package_managers.gomod.Go.release", new_callable=mock.PropertyMock)
+@mock.patch("cachi2.core.package_managers.gomod._get_gomod_version")
+@mock.patch("cachi2.core.package_managers.gomod.ModuleVersionResolver")
+@mock.patch("cachi2.core.package_managers.gomod._validate_local_replacements")
+@mock.patch("subprocess.run")
+def test_resolve_gomod_with_workspaces(
+    mock_run: mock.Mock,
+    mock_validate_local_replacements: mock.Mock,
+    mock_version_resolver: mock.Mock,
+    mock_get_gomod_version: mock.Mock,
+    mock_go_release: mock.PropertyMock,
+    cgo_disable: bool,
+    force_gomod_tidy: bool,
+    tmp_path: Path,
+    data_dir: Path,
+    gomod_request: Request,
+) -> None:
+    module_dir = gomod_request.source_dir.join_within_root("path/to/module")
+
+    # Mock the "subprocess.run" calls
+    run_side_effects = []
+
+    run_side_effects.append(
+        proc_mock(
+            "go work edit -json",
+            returncode=0,
+            stdout=get_mocked_data(data_dir, "workspaces/go_work.json"),
+        )
+    )
```

Comment:
Apart from this particular hunk, this test function is pretty much bit-for-bit identical to the existing `test_resolve_gomod` function. Given that that's the case, we should better mock `_get_go_sum_files` and add this as another parametrize case to the existing test case, this is simply a beast of a test testing a beast of a function which leads to enormous code redundancy.

Reply:
I merged the code from both tests. I needed to add some conditionals, so the test code is less readable, but probably better than the duplication we had before.

Btw, we have three boolean parameters for this test, but I don't think we need to test all combinations possible, as they shouldn't affect each other, specially considering this is a unit test and the go calls are mocked. So I added a new scenario that we have only the workspaces parameter as true, and avoided increasing the number of scenarios to 9.